src/dex/uniswap-v3/uniswap-v3-pool.ts

```diff
@@ -124,15 +124,15 @@ export class UniswapV3EventPool extends StatefulEventSubscriber<PoolState> {
     // only if the super call succeed
 
     const initPromises = [];
-    if (!this.token0sub.isInitialized) {
+    if (!this.token0sub.isInitialized && !this.dexHelper.config.isSlave) {
```

Comment:
I don't understand this part fully. Shouldn't we don't create `token0sub` and `token1sub` instances if it is slave version? It will just waste resources processing logs? Or if not subscribed, it doesn't do any work?

But later this thing will be executed:

```
    await Promise.all([
      this.token0sub.subscribeToWalletBalanceChange(
        this.poolAddress,
        blockNumber,
      ),
      this.token1sub.subscribeToWalletBalanceChange(
        this.poolAddress,
        blockNumber,
      ),
    ]);
```
Maybe exit earlier if it is not master version?

Reply:
We want to access the getBalance funciton that check redis